internal/execution/target/target.go

```diff
@@ -12,12 +15,50 @@ const (
 	TargetTypeAsync
 )
 
-type Target interface {
-	GetExecutionID() string
-	GetTargetID() string
-	IsInterruptOnError() bool
-	GetEndpoint() string
-	GetTargetType() TargetType
-	GetTimeout() time.Duration
-	GetSigningKey() string
+type Target struct {
+	ExecutionID      string              `json:"execution_id"`
+	TargetID         string              `json:"target_id"`
+	TargetType       TargetType          `json:"target_type"`
+	Endpoint         string              `json:"endpoint"`
+	Timeout          time.Duration       `json:"timeout"`
+	InterruptOnError bool                `json:"interrupt_on_error"`
+	SigningKey       *crypto.CryptoValue `json:"signing_key"`
+	SigningKeyDec    string
+}
+
+func (e *Target) GetExecutionID() string {
+	return e.ExecutionID
+}
+func (e *Target) GetTargetID() string {
+	return e.TargetID
+}
+func (e *Target) IsInterruptOnError() bool {
+	return e.InterruptOnError
+}
+func (e *Target) GetEndpoint() string {
+	return e.Endpoint
+}
+func (e *Target) GetTargetType() TargetType {
+	return e.TargetType
+}
+func (e *Target) GetTimeout() time.Duration {
+	return e.Timeout
+}
+func (e *Target) GetSigningKey(alg crypto.EncryptionAlgorithm) string {
+	if e.SigningKeyDec == "" && e.SigningKey != nil {
+		e.decryptSigningKey(alg)
+	}
+	return e.SigningKeyDec
+}
+
+func (t *Target) decryptSigningKey(alg crypto.EncryptionAlgorithm) error {
+	if t.SigningKey == nil {
+		return nil
+	}
+	keyValue, err := crypto.DecryptString(t.SigningKey, alg)
+	if err != nil {
+		return zerrors.ThrowInternal(err, "QUERY-bxevy3YXwy", "Errors.Internal")
+	}
+	t.SigningKeyDec = keyValue
+	return nil
```

Comment:
This much looks like a "Once" function with a cached output. Target is now part of the context, and context is expected to be Go routine safe. By modifying / setting a field in Target it will be a race condition.

There's two options:
- Use a [sync.OnceValues](https://pkg.go.dev/sync#OnceValues) to cache the value
- Or, don't cache the decrypted value